frontend/src/hooks/AuthProvider.jsx

```diff
@@ -1,18 +1,38 @@
-import React, { createContext } from "react";
+import React, { createContext, useContext, useEffect, useState } from "react";
 import Cookies from "js-cookie"
 import { jwtDecode } from "jwt-decode"
-export const AuthContext = createContext();
+const AuthContext = createContext();
+const useAuth = () => useContext(AuthContext);
 
-export default function AuthProvider({ children }) {
-    const token = Cookies.get("token");
-    const isLoggedIn = token ? true : false;
-    let user;
-    if (isLoggedIn){
-        user = jwtDecode(token);
+function AuthProvider({ children }) {
+    const [user, setUser] = useState(null);
+
+    useEffect(() => {
+        const token = Cookies.get("token");
+        if (token) {
+            setUser(jwtDecode(token));
+        }
+    }, []);
+
+    const loginFrontend = () => {
+        const token = Cookies.get("token");

```

Comment:
[non-blocking] Are you able to access this cookie even though it's marked as `httpOnly`?